File: official/docs/ruby/end_shipper/buy_shipment_with_endshipper.rb
require 'easypost'

EasyPost.api_key = ENV['EASYPOST_API_KEY']

shipment = EasyPost::Shipment.retrieve("shp_...")

shipment.buy(
  rate: 'rate...', 
  insurance: nil, 
  with_carbon_offset: false, 
  end_shipper_id: 'es_...'
)

puts shipment

File: official/docs/python/end_shipper/buy_shipment_with_endshipper.py
import os

import easypost


easypost.api_key = os.getenv("EASYPOST_API_KEY")

shipment = easypost.Shipment.retrieve("shp_...")

shipment.buy(
    rate="rate_...",
    end_shipper_id="es_...",
)

print(shipment)
